backend/src/org/commcare/xml/CaseXmlParser.java

```diff
@@ -184,8 +186,8 @@ public Case parse() throws InvalidStructureException, IOException, XmlPullParser
 
                     String value = parser.nextText().trim();
 
-                    if(value.equals(caseId)) {
-                        throw new InvalidStructureException("Invalid index operation! Case " + caseId + " cannot index itself!", parser);
+                    if (value.equals(caseId)) {
+                        throw new ActionableInvalidStructureException("case.error.self.index", new String[]{caseId}, "Case " + caseId + " cannot index itself");
```

Comment:
It looks like this mesage is only localized in the android repo. We should add back to the JR repos as well, right? Or were you thinking we would just use the fallback text for that? (Also fine)